tutorial/tests/test_input_output.py

```diff
@@ -86,15 +107,18 @@ def test_write_file(function_to_test, tmp_path: pl.Path):
 
 def reference_read_write_file(input_file: pl.Path, output_file: pl.Path) -> None:
     # We open the input and output file at the same time
-    with input_file.open("r") as read_file, output_file.open("w") as write_file:
-        # Here we iterate over each line of the input file
-        for line in read_file.readlines():
-            # We remove line breaks and write the line to the output file
-            clean_line = line.strip("\n\r")
-            # We crete the output line
-            output_line = f"{clean_line}, {len(clean_line)}\n"
-            # Finally we write the line and its length to the output file
-            write_file.write(output_line)
+    read_file = input_file.open("r")
+    write_file = output_file.open("w")
+    # Here we iterate over each line of the input file
+    for line in read_file.readlines():
+        # We remove line breaks and write the line to the output file
+        clean_line = line.strip("\n\r")
+        # We crete the output line
+        output_line = f"{clean_line}, {len(clean_line)}\n"
+        # Finally we write the line and its length to the output file
+        write_file.write(output_line)
+    read_file.close()
+    write_file.close()
 
 
 def test_read_write_file(function_to_test, tmp_path: pl.Path):
```

Comment:
When I tried testing this, I wrote only the strings (without removing the line endings and without writing the string length) but it would anyway pass. Then I saw that the file `lines.txt`, which is used for testing, is empty.

Reply:
I couldn't find a place where we explicitly change the lines.txt, however it is used both in test exercises as well as the explanation part and is never reset which makes it easy to change the file accidently. I took a new file now for the tests and try to reset for exercises after the write introduction.